openstudiocore/src/model/AirflowNetworkDistributionNode.cpp

```diff
@@ -0,0 +1,184 @@
+#include <utilities/idd/OS_AirflowNetworkDistributionNode_FieldEnums.hxx>
+
+#include "../utilities/units/Unit.hpp"
+
+#include "../utilities/core/Assert.hpp"
+
+namespace openstudio {
+namespace model {
+
+namespace detail {
+
+  AirflowNetworkDistributionNode_Impl::AirflowNetworkDistributionNode_Impl(const IdfObject& idfObject,
+                                                                           Model_Impl* model,
+                                                                           bool keepHandle)
+    : AirflowNetworkNode_Impl(idfObject,model,keepHandle)
+  {
+    OS_ASSERT(idfObject.iddObject().type() == AirflowNetworkDistributionNode::iddObjectType());
+  }
+
+  AirflowNetworkDistributionNode_Impl::AirflowNetworkDistributionNode_Impl(const openstudio::detail::WorkspaceObject_Impl& other,
+                                                                           Model_Impl* model,
+                                                                           bool keepHandle)
+    : AirflowNetworkNode_Impl(other,model,keepHandle)
+  {
+    OS_ASSERT(other.iddObject().type() == AirflowNetworkDistributionNode::iddObjectType());
+  }
+
+  AirflowNetworkDistributionNode_Impl::AirflowNetworkDistributionNode_Impl(const AirflowNetworkDistributionNode_Impl& other,
+                                                                           Model_Impl* model,
+                                                                           bool keepHandle)
+    : AirflowNetworkNode_Impl(other,model,keepHandle)
+  {}
+
+  const std::vector<std::string>& AirflowNetworkDistributionNode_Impl::outputVariableNames() const
+  {
+    static std::vector<std::string> result;
+    if (result.empty()){
+    }
+    return result;
+  }
+
+  IddObjectType AirflowNetworkDistributionNode_Impl::iddObjectType() const {
+    return AirflowNetworkDistributionNode::iddObjectType();
+  }
+
+  double AirflowNetworkDistributionNode_Impl::nodeHeight() const {
+    boost::optional<double> value = getDouble(OS_AirflowNetworkDistributionNodeFields::NodeHeight,true);
+    OS_ASSERT(value);
+    return value.get();
+  }
+
+  bool AirflowNetworkDistributionNode_Impl::isNodeHeightDefaulted() const {
+    return isEmpty(OS_AirflowNetworkDistributionNodeFields::NodeHeight);
+  }
+
+  void AirflowNetworkDistributionNode_Impl::setNodeHeight(double nodeHeight) {
+    bool result = setDouble(OS_AirflowNetworkDistributionNodeFields::NodeHeight, nodeHeight);
+    OS_ASSERT(result);
+  }
+
+  void AirflowNetworkDistributionNode_Impl::resetNodeHeight() {
+    bool result = setString(OS_AirflowNetworkDistributionNodeFields::NodeHeight, "");
+    OS_ASSERT(result);
+  }
+
+  void AirflowNetworkDistributionNode_Impl::resetComponent() {
+    bool result = setString(OS_AirflowNetworkDistributionNodeFields::ComponentNameorNodeName, "");
+    OS_ASSERT(result);
+  }
+
+} // detail
+
+AirflowNetworkDistributionNode::AirflowNetworkDistributionNode(const Model& model, const Handle &handle)
```

Comment:
Could make this;

    AirflowNetworkDistributionNode(const ModeObjectl& modelObject)

Reply:
I would be fine with that, your version is better. I got to this because I was fighting with Visual Studio over the legality of a similar thing to what you suggest. I gave up and went with this because it's internal and I knew it would work.